share/functions/cd.fish

```diff
@@ -3,10 +3,20 @@
 #
 function cd --description "Change directory"
     set -l MAX_DIR_HIST 25
+    set -l options h/help L P
+    argparse -n cd --max-args=3 $options -- $argv
+    or return
 
-    if test (count $argv) -gt (test "$argv[1]" = "--" && echo 2 || echo 1)
-        printf "%s\n" (_ "Too many args for cd command")
-        return 1
+    # Drop the -L argument as fish always resolves the directory. Like POSIX
+    # it's the default option for cd.
+    if set -q _flag_L
+        set argv (string replace -- '-L' '' $argv)
```

Comment:
This block isn't needed - `argparse` removes the options from `$argv`. It's a good idea to keep the comment as an explanation of the default though.